src/borrow.rs

```diff
@@ -129,25 +120,27 @@
 //!
 //! # Limitations
 //!
-//! Note that the current implementation of this is an over-approximation: It will consider all borrows potentially conflicting
-//! if the initial arrays have the same object at the end of their [base object chain][base].
-//! For example, creating two views of the same underlying array by slicing will always yield potentially conflicting borrows
-//! even if the slice indices are chosen so that the two views do not actually share any elements by splitting the array into
-//! non-overlapping parts of by interleaving along one of its axes.
+//! Note that the current implementation of this is an over-approximation: It will consider borrows
+//! potentially conflicting if the initial arrays have the same object at the end of their [base object chain][base].
+//! Then, multiple conditions which are sufficient but not necessary to show the absence of conflicts are checked,
+//! but there are cases which they do not handle, for example slicing an array with a step size
+//! that does not divide its dimension along that axis. In these situations, borrows are rejected even though the arrays
+//! do not actually share any elements.
 //!
 //! This does limit the set of programs that can be written using safe Rust in way similar to rustc itself
 //! which ensures that all accepted programs are memory safe but does not necessarily accept all memory safe programs.
-//! The plan is to refine this checking to correctly handle more involved cases like non-overlapping and interleaved
-//! views into the same array and until then the unsafe method [`PyArray::as_array_mut`] can be used as an escape hatch.
+//! The plan is to refine this checking to correctly handle more involved cases and until then
```

Comment:
Rather than committing to "a plan" I would suggest the less concrete "In the future more complex cases may be handled".